js/generate.js

```diff
@@ -66,22 +67,38 @@ function maturityData(spec) {
 	maturity = {label: spec.maturity, level:level};
     } else {
 	maturity = {label:spec.maturity, level: maturityLevels[spec.maturity]};
-	maturityIcon = {src:"http://www.w3.org/2013/09/wpd-rectrack-icons/" + spec.maturity.toLowerCase().replace(/lastcall/,'lcwd') + '.svg', alt:spec.maturity, width:50, height:50};
-	if (spec.maturity == "REC" || spec.maturity == "LastCall") {
+	maturityIcon = {src:"http://www.w3.org/2013/09/wpd-rectrack-icons/" + maturityMapping[spec.maturity].toLowerCase() + '.svg', alt:spec.maturity, width:50, height:50};
+	if (spec.maturity == "Recommendation" || spec.maturity == "Last Call") {
 	    maturityIcon.height = 53;
 	}
     }
     return {maturity: maturity, maturityIcon: maturityIcon};
 }
 
-var specData;
-var specXhr = new XMLHttpRequest();
-specXhr.open("GET", "specs/tr.json");
-specXhr.onload = function() {
-    specData = JSON.parse(this.responseText);
-    fillTables();
-};
-specXhr.send();
+var specData = {};
+fillTables();
+var w3cApiAuth = "?apikey=rvl38z1hpvkw0g804gk48o0s8408w88";
+var w3cApiFetchConfig = { method:'GET', mode:'cors'};
+
+
+function fetchSpecData(d) {
+    if (!d.TR) {
+        return new Promise((res,rej) => { res({maturity:d.maturity, wgs: d.wgs, title: d.title});});
+    }
+    var shortName = d.TR.split('#')[0].split('/')[4];
+    var ret = {};
+    return fetch("https://api.w3.org/specifications/" + shortName + "/versions" + w3cApiAuth + "&embed=1", w3cApiFetchConfig)
+        .then(response =>  response.json())
+        .then(function(data) {
+            var latest = data._embedded['version-history'][0];
+            ret = { maturity: latest.status, title: latest.title};
+            return fetch(latest._links["deliverers"].href  + w3cApiAuth + "&embed=1", w3cApiFetchConfig).then(response => response.json());
+        })
+        .then(function(data) {
+            ret["wgs"] = data._embedded.deliverers.map(g => ({label: g.name, url: g._links["homepage"].href}));
```

Comment:
Nit: you are accessing a "_private" prop? Maybe expose it as public on whatever g is?